src/GraphControl/Control/Grapher.cpp

```diff
@@ -1088,10 +1108,23 @@ optional<vector<shared_ptr<Graphing::IEquation>>> Grapher::TryInitializeGraph(bo
 {
     if (keepCurrentView || IsKeepCurrentView)
     {
+        auto renderer = m_graph->GetRenderer();
         double xMin, xMax, yMin, yMax;
-        m_graph->GetRenderer()->GetDisplayRanges(xMin, xMax, yMin, yMax);
+        renderer->GetDisplayRanges(xMin, xMax, yMin, yMax);
         auto initResult = m_graph->TryInitialize(graphingExp);
-        m_graph->GetRenderer()->SetDisplayRanges(xMin, xMax, yMin, yMax);
+        if (initResult != nullopt)
+        {
+            if (IsKeepCurrentView)
+            {
+                if (SUCCEEDED(renderer->PrepareGraph()))
```

Comment:
Do you know what this function is doing or when it can fail? A comment explaining would be useful. Also, is the code path for when this fails valid? Will the below call to renderer->SetDisplayRanges crash? Is an error shown on the last added equation?

Reply:
I've asked the graphing engine team what happens. I'm waiting to hear back.

Reply:
The graphing engine team got back to me and said that PrepareGraph should have no affect on any other API calls, so we are safe to call SetDisplayRanges if PrepareGraph fails.